src/corebluetooth/device.rs

```diff
@@ -31,7 +35,13 @@ impl Eq for DeviceImpl {}
 
 impl std::hash::Hash for DeviceImpl {
     fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
-        self.peripheral.hash(state);
+        self.peripheral
+            .dispatch(|peripheral| {
+                let mut state = std::hash::DefaultHasher::new();
+                peripheral.hash(&mut state);
+                state.finish()
+            })
+            .hash(state)
```

Comment:
This can go back to just `self.peripheral.hash(state)`. I initially didn't have a `Hash` impl on `Dispatched`, but that's fixed now.